src/Downloader.ts

```diff
@@ -184,37 +184,56 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
+  public setBaseUrls(baseUrl: string, baseUrlForMainPage: string = this.mw.veApiUrl.href){
+    this.baseUrl = baseUrl;
+    this.baseUrlForMainPage = baseUrlForMainPage;
+  }
+
+  public async checkRestApiAvailabilty(type: string, restApiBase: string, defaultUrlForMainPage: string){
```

Comment:
I believe you can make this method more generic (based on HTTP response status code), returning a boolean, called `checkApiAvailability(url)`